spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMetadataTables.java

```diff
@@ -153,7 +155,10 @@ public void testUnpartitionedTable() throws Exception {
     assertThat(expectedDataManifests).as("Should have 1 data manifest").hasSize(1);
     assertThat(expectedDeleteManifests).as("Should have 1 delete manifest").hasSize(1);
 
-    Schema entriesTableSchema = Spark3Util.loadIcebergTable(spark, tableName + ".entries").schema();
+    Schema entriesTableSchema =
+        TypeUtil.selectNot(
+            Spark3Util.loadIcebergTable(spark, tableName + ".entries").schema(),
+            Set.of(DataFile.FIRST_ROW_ID.fieldId()));
```

Comment:
The assertions in this suite were previously reading the manifest directly as Avro and matching against the rows returned by Spark. That always produces `null` for `first_row_id` but the manifest reader sets the field using inheritance so the tests were failing.

The solution is to remove `first_row_id` from both the entries table schema (used to read manifests) and from the Spark datasets (via `selectNonDerived`).

Reply:
May be worth adding a
```java
private Schema entriesTableSchema() {
        TypeUtil.selectNot(
            Spark3Util.loadIcebergTable(spark, tableName + ".entries").schema(),
            Set.of(DataFile.FIRST_ROW_ID.fieldId()));
}
```

To reduce the redundancy